### PR TITLE
[#194] Update to new dependencies and GHC 9

### LINKED ISSUE #194: Update to new dependencies and GHC 9
# Clarification and motivation

New versions have been released of `stablecoin`'s dependencies, in particular:
- [morley-1.17.0](https://hackage.haskell.org/package/morley-1.17.0)
- [lorentz-0.14.0](https://hackage.haskell.org/package/lorentz-0.14.0)
- [morley-client-0.2.0](https://hackage.haskell.org/package/morley-client-0.2.0)
- [cleveland-0.2.0](https://hackage.haskell.org/package/cleveland-0.2.0)

moreover:
1. one for `indigo` should be released after [indigo/#58](https://gitlab.com/morley-framework/indigo/-/issues/58 )
2. `morley-ledgers` will be up-to-date after [morley-ledgers/#25](https://gitlab.com/morley-framework/morley-ledgers/-/issues/25)
3. `morley-metadata` will be up-to-date after [morley-metadata/#15](https://gitlab.com/morley-framework/morley-metadata/-/issues/15)

_Note: this issue depends on these ones_

Consequently `stablecoin` should be updated to use them and switch to GHC 9 

It would also be good to solve https://github.com/tqtezos/stablecoin/issues/185 first and tag a new release

# Acceptance criteria

1. The project is updated to support and use GHC 9 as well as the new versions of dependencies mentioned above.
2. The version number is bumped in preparation for a new release

### PR DESCRIPTION
**Pull request by dcastro**
## Description

This PR bumps the morley & friends dependencies to their latest versions.

It also fixes a couple of small things we missed in #134 and #135.

## Related issue(s)

Resolves #194

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock
  - [x] I updated [changelog](../tree/master/ChangeLog.md) unless I am sure my changes are
        not essential.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: ligo/stablecoin/fa1.2/actions.ligo [hunk 1]
```diff
@@ -681,16 +681,19 @@ function add_permit
...
+; const updated_store : storage = sender_check(owner, store, full_param, "NOT_PERMIT_ISSUER")
+
 ; const updated_permits : permits =
-    case param.1 of [
+    case param.permit_hash of [
     | None ->
-        set_user_default_expiry(Tezos.get_sender(), new_expiry, store.permits)
-    | Some(hash_and_address) ->
-        set_permit_expiry(hash_and_address.1, hash_and_address.0, new_expiry, store.permits, store.default_expiry)
+        set_user_default_expiry(owner, param.expiry, updated_store.permits)
```
Comment (sras) [inline ligo/stablecoin/fa1.2/actions.ligo:692]: It seems that the call to `set_user_default_expiry` is made with `param.issuer` while previously it was called with `Tezos.get_sender()`. Is this change intentional?

Comment (dcastro) [reply ligo/stablecoin/fa1.2/actions.ligo:692]: >  Is this change intentional?

Yes, that was the same change we applied to `actions.ligo` for the FA2 variant of the contract in #134 (see the diff [here](https://github.com/tqtezos/stablecoin/pull/134/files#diff-3f97ef02a46298c9eb30b26bd76a27da8fea70f49db7d54799b3c814eda91357L761-R768)), but back then we forgot to apply the same changes to the FA1.2 variant. 

The description of the first commit in this PR contains some details about this as well.

**Review by sras — APPROVED** [review]
LGTM
